pymongo/errors.py

```diff
@@ -145,9 +160,9 @@ class NotPrimaryError(AutoReconnect):
     .. versionadded:: 3.12
     """
 
-    def __init__(
-        self, message: str = "", errors: Optional[Union[Mapping[str, Any], List]] = None
-    ) -> None:
+    details: _DocumentOut
+
+    def __init__(self, message: str = "", errors: Optional[_DocumentOut] = None) -> None:
```

Comment:
Why did we need to change this from Mapping to _DocumentOut?

Reply:
If i recall correctly, this had to do with the change below. since i'm undoing that change, this is also Mapping again